src/shared/identityUtils.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import fs from 'node:fs';
+import { randomBytes } from 'node:crypto';
+import { DevServerUtils } from './devServerUtils.js';
+
+class LwrConfigFile {
+  public identityToken?: string;
+}
+
+const BYTES_TO_GENERATE = 32;
+
+export class IdentityUtils {
+  public static async createIdentityToken(): Promise<void> {
+    const lwrConfigFile = DevServerUtils.getServerConfigFileLocation();
+    if (fs.existsSync(lwrConfigFile)) {
+      const config = DevServerUtils.fetchServerConfigFileContent() as LwrConfigFile;
+      if (config?.identityToken == null) {
```

Comment:
Again as stated above, if for some reason we cannot read the content of the config file (i.e `config==null || config==undefined) then that should be an error case. So maybe update the code here like:
```
  public static async createIdentityToken(byteSize = 32): Promise<void> {
    const config = await DevServerUtils.fetchServerConfigFileContent() as LwrConfigFile;
    if (config && !config.identityToken) {
        config.identityToken = randomBytes(byteSize).toString();
        await DevServerUtils.writeServerConfigFileContent(config);
    }
  }
```

This way, if any of the `await ...` statements throws an error (config file not found, unable to read/write config file, etc) then the exception will be bubbled up to the caller (without wrapping it in another Error obj) and the code would be simpler?

Reply:
Also maybe we should rename the method to something that is more descriptive of what it is doing? Since the method will generate token and save to file, maybe rename to `updateServerConfigFileWithIdentityToken` ?